Data/data_augmentation.py

```diff
@@ -0,0 +1,112 @@
+import os
+import random
+from scipy import ndarray
+
+# image processing library
+import skimage as sk
+from skimage import transform
+from skimage import util
+from skimage import io
+from skimage import data, exposure
+
+#rotation
+
+def vc_rotation(image_array: ndarray):
+    # pick a random degree of rotation between 25% on the left and 25% on the right
+    random_degree = random.uniform(-25, 25)
+    return sk.transform.rotate(image_array, random_degree)
+
+
+#noise
+
+def vc_noise(image_array: ndarray):
+    # add random noise to the image
+    return sk.util.random_noise(image_array)
+
+#brightness
+
+def vc_brightness(image_array: ndarray):
+
+    gamma=1
+    gain=1
+
+    if is_random:
+        gamma = np.random.uniform(1-gamma, 1+gamma)
+    return exposure.adjust_gamma(image_array, gamma, gain)
+   
+
+#crop
+
+def vc_crop(image_array: ndarray):
+    """Pad image by n_pixels on each size, then take random crop of same
+    original size.
```

Comment:
What is n_pixels?

Reply:
if we are not assuming , its a square image , we are padding it with some pixels.


or we can use this to change the height and width,
from skimage.util import crop


# crop_width{sequence, int}: Number of values to remove from the edges of each axis. 
# ((before_1, after_1), … (before_N, after_N)) specifies unique crop widths at the 
# start and end of each axis. ((before, after),) specifies a fixed start and end 
# crop for every axis. (n,) or n for integer n is a shortcut for before = after = n 
# for all axes.
B = crop(A, ((50, 100), (50, 50), (0,0)), copy=False)

print(A.shape, B.shape)
# (220, 220, 3) (70, 120, 3)